### PR TITLE
Busybox 1.24.1

### PR DESCRIPTION
**Pull request by sknsean**

### CODE REVIEW

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 1]
```diff
@@ -169,14 +185,30 @@ CONFIG_DATE=y
...
+# CONFIG_DD is not set
+# CONFIG_FEATURE_DD_SIGNAL_HANDLING is not set
+# CONFIG_FEATURE_DD_THIRD_STATUS_LINE is not set
+# CONFIG_FEATURE_DD_IBS_OBS is not set
+# CONFIG_FEATURE_DD_STATUS is not set
+# CONFIG_HOSTID is not set
+# CONFIG_ID is not set
+# CONFIG_GROUPS is not
 # CONFIG_SHUF is not set
+CONFIG_SYNC=
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Hmm...  Is this a mistake?

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 2]
```diff
@@ -169,14 +185,30 @@ CONFIG_DATE=y
 # CONFIG_FEATURE_DATE_ISOFMT is not set
 # CONFIG_FEATURE_DATE_NANO is not set
 # CONFIG_FEATURE_DATE_COMPAT is not set
+# CONFIG_DD is not set
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: You are disabling dd, which was disabled in the old recipe.  Is that on purpose?

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 3]
```diff
@@ -169,14 +185,30 @@ CONFIG_DATE=y
...
+# CONFIG_FEATURE_DD_SIGNAL_HANDLING is not set
+# CONFIG_FEATURE_DD_THIRD_STATUS_LINE is not set
+# CONFIG_FEATURE_DD_IBS_OBS is not set
+# CONFIG_FEATURE_DD_STATUS is not set
+# CONFIG_HOSTID is not set
+# CONFIG_ID is not set
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: CONFIG_ID was enabled in previous version.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 4]
```diff
@@ -263,16 +289,15 @@ CONFIG_FEATURE_FLOAT_SLEEP=y
 # CONFIG_FEATURE_STAT_FORMAT is not set
 CONFIG_STTY=y
 # CONFIG_SUM is not set
-CONFIG_SYNC=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Oh, additional comment to the strange CONFIG_SYNC line above.  It should stay enabled :)

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 5]
```diff
@@ -512,10 +548,29 @@ CONFIG_DEFAULT_DEPMOD_FILE="modules.dep"
...
 # CONFIG_BLOCKDEV is not set
 # CONFIG_FATATTR is not set
+# CONFIG_FSTRIM is not set
+# CONFIG_MDEV is not set
+# CONFIG_FEATURE_MDEV_CONF is not set
+# CONFIG_FEATURE_MDEV_RENAME is not set
+# CONFIG_FEATURE_MDEV_RENAME_REGEXP is not set
+# CONFIG_FEATURE_MDEV_EXEC is not set
+# CONFIG_FEATURE_MDEV_LOAD_FIRMWARE is not set
+CONFIG_MOUNT=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: This was disabled by default before, and a USE flag exist to enable it. This change thus both changes the default and makes the flag not have any effect...

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 6]
```diff
@@ -689,28 +744,37 @@ CONFIG_MOUNTPOINT=y
...
+# CONFIG_FEATURE_NAMEIF_EXTENDED is not set
 # CONFIG_NBDCLIENT is not set
 # CONFIG_NC is not set
 # CONFIG_NC_SERVER is not set
 # CONFIG_NC_EXTRA is not set
 # CONFIG_NC_110_COMPAT is not set
+# CONFIG_PING is not set
+# CONFIG_PING6 is not set
+# CONFIG_FEATURE_FANCY_PING is not set
+CONFIG_WGET=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Again, this was disabled by default before, and a USE flag exists to control it.  You are breaking the USE flag, and changing defaults.  Don't.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 7]
```diff
@@ -689,28 +744,37 @@ CONFIG_MOUNTPOINT=y
...
+# CONFIG_FEATURE_WGET_STATUSBAR is not set
+# CONFIG_FEATURE_WGET_AUTHENTICATION is not set
+# CONFIG_FEATURE_WGET_LONG_OPTIONS is not set
+# CONFIG_FEATURE_WGET_TIMEOUT is not set
+CONFIG_FEATURE_WGET_OPENSSL=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: I don't think we want to hardcode requirements for OpenSSL in busybox.  Sort of defeats the purpose of staying really small.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 8]
```diff
@@ -759,7 +823,7 @@ CONFIG_FEATURE_IFUPDOWN_IP=y
 CONFIG_FEATURE_IFUPDOWN_IPV4=y
 # CONFIG_FEATURE_IFUPDOWN_IPV6 is not set
 # CONFIG_FEATURE_IFUPDOWN_MAPPING is not set
-CONFIG_FEATURE_IFUPDOWN_EXTERNAL_DHCP=y
+# CONFIG_FEATURE_IFUPDOWN_EXTERNAL_DHCP is not set
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Why are you changing this?  I think you are causing breakage in the busybox_udhcp USE flag.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 9]
```diff
@@ -783,18 +848,13 @@ CONFIG_FEATURE_IP_ROUTE=y
...
 # CONFIG_FEATURE_IPCALC_LONG_OPTIONS is not set
-# CONFIG_NAMEIF is not set
-# CONFIG_FEATURE_NAMEIF_EXTENDED is not set
 CONFIG_NETSTAT=y
 # CONFIG_FEATURE_NETSTAT_WIDE is not set
 CONFIG_FEATURE_NETSTAT_PRG=y
 CONFIG_NSLOOKUP=y
 # CONFIG_NTPD is not set
 # CONFIG_FEATURE_NTPD_SERVER is not set
-CONFIG_PING=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: You are disabling ping.  Probably a good idea, but...  It should go in separate commit, and you should provide a USE flag to enable it.  Or you or someone else should come up with a better plan for how to control this busybox spaghetti...

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 10]
```diff
@@ -884,20 +955,13 @@ CONFIG_KILLALL5=y
 # CONFIG_PKILL is not set
 CONFIG_PS=y
 # CONFIG_FEATURE_PS_WIDE is not set
+CONFIG_FEATURE_PS_LONG=y
 # CONFIG_FEATURE_PS_TIME is not set
 # CONFIG_FEATURE_PS_ADDITIONAL_COLUMNS is not set
 # CONFIG_FEATURE_PS_UNUSUAL_SYSTEMS is not set
 # CONFIG_RENICE is not set
 CONFIG_BB_SYSCTL=y
-CONFIG_TOP=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Disabling top, which was enabled in previous version. Why?

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 11]
```diff
@@ -884,20 +955,13 @@ CONFIG_KILLALL5=y
...
-CONFIG_FEATURE_TOP_CPU_USAGE_PERCENTAGE=y
-CONFIG_FEATURE_TOP_CPU_GLOBAL_PERCENTS=y
-# CONFIG_FEATURE_TOP_SMP_CPU is not set
-CONFIG_FEATURE_TOP_DECIMALS=y
-# CONFIG_FEATURE_TOP_SMP_PROCESS is not set
-CONFIG_FEATURE_TOPMEM=y
 # CONFIG_FEATURE_SHOW_THREADS is not set
-CONFIG_UPTIME=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Disabling uptime, which was enabled in previous version. Why?

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 12]
```diff
@@ -966,15 +1032,16 @@ CONFIG_UPTIME=y
 # CONFIG_MSH is not set
 # CONFIG_FEATURE_SH_IS_ASH is not set
 # CONFIG_FEATURE_SH_IS_HUSH is not set
-CONFIG_FEATURE_SH_IS_NONE=y
+# CONFIG_FEATURE_SH_IS_NONE is not set
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Why?

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 13]
```diff
@@ -966,15 +1032,16 @@ CONFIG_UPTIME=y
...
 # CONFIG_FEATURE_BASH_IS_ASH is not set
 # CONFIG_FEATURE_BASH_IS_HUSH is not set
-CONFIG_FEATURE_BASH_IS_NONE=y
+# CONFIG_FEATURE_BASH_IS_NONE is not set
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Why?

**Comment by sknsean** [comment]
Just made an make oldconfig for busybox and i think i got it all :+1:

**Comment by sknsean** [comment]
It is not very easy to diff the two defconfigs because a lot of thinks have been moved around...

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 14]
```diff
@@ -554,24 +608,9 @@ CONFIG_HEXDUMP=y
...
-# CONFIG_FEATURE_MDEV_RENAME is not set
-# CONFIG_FEATURE_MDEV_RENAME_REGEXP is not set
-# CONFIG_FEATURE_MDEV_EXEC is not set
-# CONFIG_FEATURE_MDEV_LOAD_FIRMWARE is not set
 # CONFIG_MKSWAP is not set
 # CONFIG_FEATURE_MKSWAP_UUID is not set
 CONFIG_MORE=y
-# CONFIG_MOUNT is not set
-# CONFIG_FEATURE_MOUNT_FAKE is not set
-CONFIG_FEATURE_MOUNT_VERBOSE=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: You should take care to keep all these CONFIG_FEATURE_MOUNT_* options in.

Or better, fixup the busybox_mount USE flag specified in busybox-configure.inc, to enable them that way.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 15]
```diff
@@ -664,15 +727,6 @@ CONFIG_INOTIFYD=y
 CONFIG_LAST=y
 CONFIG_FEATURE_LAST_SMALL=y
 # CONFIG_FEATURE_LAST_FANCY is not set
-# CONFIG_LESS is not set
-CONFIG_FEATURE_LESS_MAXLINES=9999999
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: You should leave all these CONFIG_FEATURE_LESS_* options as they were, or better, add them to the busybox_less USE flag.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 16]
```diff
@@ -768,9 +831,10 @@ CONFIG_FEATURE_IFUPDOWN_EXTERNAL_DHCP=y
 # CONFIG_FEATURE_INETD_SUPPORT_BUILTIN_CHARGEN is not set
 # CONFIG_FEATURE_INETD_RPC is not set
 # CONFIG_IP is not set
-CONFIG_FEATURE_IP_ADDRESS=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Leave these enabled, or fixup the busybox_ip USE flag.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 17]
```diff
@@ -189,10 +220,6 @@ CONFIG_CHOWN=y
 CONFIG_CP=y
 # CONFIG_FEATURE_CP_LONG_OPTIONS is not set
 CONFIG_CUT=y
-CONFIG_DD=y
-CONFIG_FEATURE_DD_SIGNAL_HANDLING=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Another flag you seem to have disabled by mistake.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 18]
```diff
@@ -189,10 +220,6 @@ CONFIG_CHOWN=y
...
-# CONFIG_FEATURE_DD_THIRD_STATUS_LINE is not set
-CONFIG_FEATURE_DD_IBS_OBS=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Another flag you seem to have disabled by mistake.

File: recipes/busybox/busybox-1.24.1/defconfig [hunk 19]
```diff
@@ -783,18 +847,13 @@ CONFIG_FEATURE_IP_ROUTE=y
...
-# CONFIG_FEATURE_NAMEIF_EXTENDED is not set
 CONFIG_NETSTAT=y
 # CONFIG_FEATURE_NETSTAT_WIDE is not set
 CONFIG_FEATURE_NETSTAT_PRG=y
 CONFIG_NSLOOKUP=y
 # CONFIG_NTPD is not set
 # CONFIG_FEATURE_NTPD_SERVER is not set
-CONFIG_PING=y
-# CONFIG_PING6 is not set
-CONFIG_FEATURE_FANCY_PING=y
```
Comment (esben) [inline recipes/busybox/busybox-1.24.1/defconfig]: Another flag you seem to have disabled by mistake.

**Comment by sknsean** [comment]
I have checked it again with https://raw.githubusercontent.com/l0b0/diff-ignore-moved-lines/master/diff-ignore-moved-lines.sh :-)
We need to investigate these options:
CONFIG_PID_FILE_PATH="/var/run"
CONFIG_SYSROOT=""
CONFIG_EXTRA_LDFLAGS=""
CONFIG_EXTRA_LDLIBS=""
CONFIG_UNAME_OSNAME="GNU/Linux" ??? OE-Lite??
CONFIG_FEATURE_CROND_DIR=""
CONFIG_LAST_ID="0" ?
CONFIG_FEATURE_DEFAULT_PASSWD_ALGO="des" okay??
CONFIG_FEATURE_IP_ROUTE_DIR=""

**Comment by esben** [comment]
I think we should keep "GNU/Linux" as CONFIG_UNAME_OSNAME.

**Comment by kimrhh** [comment]
i think you should change this in two ways:

1: only add new options to the oe-lite defconfig, do not change existing ones
2: add a seperate upstream_defconfig and make a USE to be able to build with that. That way the next busybox bump will be much easier

**Comment by sknsean** [comment]
How have solution 1 in this PR :-)

**Comment by esben** [comment]
Could someone please explain what the purpose of the new default_defconfig file is?

**Comment by kimrhh** [comment]
to diff the default config when it is changing

**Comment by esben** [comment]
Could you add some explanation on exactly how you envision this file being used?  Otherwise, I don't think it will make any difference...
